pkg/operator/controller/controller.go

```diff
@@ -129,19 +129,21 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 
 			if err := r.enforceEffectiveIngressDomain(ingress, ingressConfig); err != nil {
 				errs = append(errs, fmt.Errorf("failed to enforce the effective ingress domain for clusteringress %s: %v", ingress.Name, err))
-			} else if err := r.enforceEffectiveEndpointPublishingStrategy(ingress, infraConfig); err != nil {
-				errs = append(errs, fmt.Errorf("failed to enforce the effective HA configuration for clusteringress %s: %v", ingress.Name, err))
-			} else if ingress.DeletionTimestamp != nil {
-				// Handle deletion.
-				if err := r.ensureIngressDeleted(ingress, dnsConfig, infraConfig); err != nil {
-					errs = append(errs, fmt.Errorf("failed to ensure ingress deletion: %v", err))
-				}
-			} else if err := r.enforceIngressFinalizer(ingress); err != nil {
-				errs = append(errs, fmt.Errorf("failed to enforce ingress finalizer %s/%s: %v", ingress.Namespace, ingress.Name, err))
-			} else {
-				// Handle everything else.
-				if err := r.ensureClusterIngress(ingress, dnsConfig, infraConfig); err != nil {
-					errs = append(errs, fmt.Errorf("failed to ensure clusteringress: %v", err))
+			} else if len(ingress.Status.Domain) > 0 {
+				if err := r.enforceEffectiveEndpointPublishingStrategy(ingress, infraConfig); err != nil {
+					errs = append(errs, fmt.Errorf("failed to enforce the effective HA configuration for clusteringress %s: %v", ingress.Name, err))
+				} else if ingress.DeletionTimestamp != nil {
+					// Handle deletion.
+					if err := r.ensureIngressDeleted(ingress, dnsConfig, infraConfig); err != nil {
+						errs = append(errs, fmt.Errorf("failed to ensure ingress deletion: %v", err))
+					}
+				} else if err := r.enforceIngressFinalizer(ingress); err != nil {
+					errs = append(errs, fmt.Errorf("failed to enforce ingress finalizer %s/%s: %v", ingress.Namespace, ingress.Name, err))
+				} else {
+					// Handle everything else.
+					if err := r.ensureClusterIngress(ingress, dnsConfig, infraConfig); err != nil {
+						errs = append(errs, fmt.Errorf("failed to ensure clusteringress: %v", err))
+					}
```

Comment:
@Miciah PTAL, I think you had some ideas about how to tame this control flow — it's getting out of hand IMO

Reply:
At one time I suggested using a switch, but that in fact would not work (we cannot do `case x := y; y != nil:`).  I think what we need to do is factor out a couple functions, such as `getClusterConfig` to get the cluster config and `reconcileIngressController` to ensure defaults and then create or delete resources as needed.  I can make a separate PR with what I have in mind if that would help.